File: Data Structures & Algorithms/daily-temperatures/submission-1.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        std::stack<std::pair<int, int>> dec_temp;
        size_t n = temperatures.size();
        std::vector<int> res(n, 0);
        for (size_t i = 0; i < n; i++) {
            while (!dec_temp.empty() && temperatures[i] > dec_temp.top().second) {
                auto [idx, _] = dec_temp.top();
                res[idx] = i - idx;
                dec_temp.pop();
            }
            dec_temp.push(std::make_pair(i, temperatures[i]));
        }
        return res;
    }
};

File: Data Structures & Algorithms/pow-x-n/submission-1.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0:
            return 1.0
        negative = n < 0
        n = abs(n)
        stack = []
        res = x
        while n > 1:
            rem = n % 2
            n //= 2
            if rem:
                stack.append(res)
            res *= res
        while stack:
            res *= stack.pop()
        return 1/res if negative else res